### ISSUE TITLE
Generated swagger for nested isolated applications

### ISSUE THREAD

**Issue by vdecristofaro** [domaindrivendev/Swashbuckle.AspNetCore#1024]
I am organizing my projects to use "nested applications" as described in this [blog post](http://dhickey.ie/2018/06/09/aspnet-core-nested-apps/) from Damian Hickey.
Those nested applications are in satellite net standard 2.0 library.

Routing Is correctly configured with path segment and assume the following template
https://server/resource/{version}/route

Applications in the satellite libraries are completely isolated from the main host and I am able to configure different versioning, different mvc pipelines, even different mediators. Instances of the same service behave differently in every different nested app.

I am using the following sintax to isolate those applications into dedicated paths

```
app.IsolatedMap<NestedStartup1>( "/app1" );
app.IsolatedMap<NestedStartup2>( "/app2" );
```

As a side issue, because the apps are completely isolated the swagger documentation is generated only for the main application and for the nested, each of them in a different path. For example:

1. Main documentation is accessible at https://localhost:5001/swagger
2. Documentation for nested 1 is accessible at https://localhost:5001/app1/swagger
3. Documentation for nested 2 is accessible at https://localhost:5001/app2/swagger

I have the opportunity the share interfaces between the main application and the nested, as it already happen for `IHttpContextAccessor` for example. Does anyone know which interface should I share to make all the documentation being generated together?

Reproducible code sample available [here](https://github.com/vdecristofaro/ApiVersioningNestedApps).

**Comment by domaindrivendev** [comment]
I'm not overly familiar with the setup you've described but I can tell you that SB leverages `IApiDescriptionGroupCollectionProvider` to generate Operation descriptions. If you can somehow configure it's implementation to incorporate _all_ actions across your nested apps then that might help

**Comment by vdecristofaro** [comment]
Thanks for your answer.
I have been able to flow the `IApiDescriptionGroupCollectionProvider` instance created in the host application into the nested apps. 
Inside the nested applications I removed the call to `AddApiExplorer` and `AddVersionedApiExplorer` and substituted these with only the following

`services.TryAddEnumerable( ServiceDescriptor.Transient<IApiDescriptionProvider, DefaultApiDescriptionProvider>() );`

however the swagger does not see nested applications routes. 

Who is responsible for populating `IApiDescriptionGroupCollectionProvider` with route data and when this happens?
